### ISSUE TITLE
🌟Feature request: Add more options to `closeOnEscape` prop

### ISSUE THREAD

**Issue by anatolzak** [melt-ui/melt-ui#1141]
**Is your feature request related to a problem? Please describe.**
When dealing with multiple nested floating elements and passing `closeOnEscape: false` on a child element, when the escape key is pressed, both the parent and the child close. I would expect neither the parent nor the child to close since I explicitly "asked" the child not to close on escape

**Describe the solution you'd like**
We should be able to: 
- pass `closeOnEscape: null` to specify that the parent floating element should be responsible when escape is pressed.
- pass `closeOnEscape: false` to specify that the child and any parents should not close when escape is pressed.

**Describe alternatives you've considered**

**Additional context**

**Comment by huntabyte** [comment]
I feel like this is something we should be able to tackle internally so the user shouldn't have to think about it. Does the current stack setup in #1080 not support this behavior?

**Comment by anatolzak** [comment]
> Also, turn on GitHub sponsors or drop a donate link so I can send you something for all the incredible work you've been doing recently 😃

@huntabyte I really appreciate that Hunter! It has been super fun doing all this! I submitted my application for Github sponsors and it's being reviewed now.



> Something along the lines of: "If I'm at the top of the stack and my `closeOnEscape` prop is `false`, don't do anything".

That's what https://github.com/melt-ui/melt-ui/pull/1080 implements. However, I am suggesting an additional option where the component is not responsible for the escape key press. 

For example if you want to have an open tooltip inside a dialog and that when escape is pressed, the dialog closes. Currently with https://github.com/melt-ui/melt-ui/pull/1080, the options are to either only close the tooltip on escape using `closeOnEscape: true` or to not close the tooltip on escape using `closeOnEscape: false`. There is no option for tooltip to ignore the escape and let it "bubble" to the underneath layers. 

Before https://github.com/melt-ui/melt-ui/pull/1080, when a tooltip was open inside a dialog and the tooltip has `closeOnEscape: false`, pressing escape would close the dialog even though the tooltip "asked" to not close when escape is pressed. But then there is no way to disable an escape while the tooltip is open without it cascading to the underneath layers.

Basically, what I am suggesting is that we add 3 options for `closeOnEscape`:
- `true`: on escape, closes the element
- `false`: on escape, doesn't close anything
- `null`, doesn't do anything to the current element and let's the event cascade to previous layers. And so if a previous layer has `closeOnEscape: true`, the previous layer will close which will then also close the top-most element.
